mllib-local/src/main/scala/org/apache/spark/ml/linalg/Vectors.scala

```diff
@@ -504,6 +506,10 @@ object Vectors {
 
   /** Max number of nonzero entries used in computing hash code. */
   private[linalg] val MAX_HASH_NNZ = 128
+
+  private[ml] def getSparseSize(nnz: Long): Long = nnz * 12 + 20
```

Comment:
Let's try to avoid hard coding constants.

Reply:
SG